README.md

```diff
@@ -223,7 +223,79 @@ _You do not need to commit the generated `./doc` or `.yardoc` files._
   end
 ```
 
-- The `and` and `or` keywords are banned. It's just not worth it. Always use `&&` and `||` instead.
+- Do not use `and` and `or` in boolean context - `and` and `or` are control flow operators and should be used as such. They have very low precedence, and can be used as a short form of specifying flow sequences like "evaluate expression 1, and only if it is not successful (returned `nil`), evaluate expression 2". This is especially useful for raising errors or early return without breaking the reading flow.
+
+```ruby
+# good: and/or for control flow
+x = extract_arguments or raise ArgumentError, "Not enough arguments!"
+user.suspended? and return :denied
```

Comment:
Would we want to use
```ruby
user.suspended? and return :denied
```
instead of 
```ruby
return :denied if user.suspended?
```

How do we choose? A bit wary of opening up a "two ways to do one thing" situation, when the existing codebase is consistently using a form that looks just as powerful.

Reply:
I would go with the `return...if` if that case. I do think think this is a poor example when you could use a guard if instead.

As you say, I think and/or work better when you're checking whether an action was successful. You can achieve the same thing with if/unless but I find that is actually less readable in the action case:

```ruby
do_something or return :failure

return :failure unless do_something
```

Happy to modify the examples if you think that pattern makes sense.

Reply:
In our examples above, `do_something` needs to return a truth-y value, but does not guarantee a boolean as `do_something?` would.

Are we putting a invisible dependency on the called method then, and risk someone changing `do_something` in a perfectly valid way, which breaks the implicit truth-y-ness return requirement? Is that an actual risk, and is it acceptable in respect to what we gain from relaxing the rules around `and`/`or`?

(Maybe the above is not very clear, I'll try again with a fresh brain tomorrow)